### PR TITLE
Simplify the implementation

### PR DESCRIPTION
**Pull request by nathansobo**
https://github.com/atom/text-buffer/pull/132 upgrades Atom's `text-buffer` module to Jasmine 2.4 and enables us to use Babel. This is a big improvement, but one frequent need, especially with `text-buffer`, is the ability to run tests inside an Electron window. Previously, we were able to run `text-buffer`'s tests as if it were an Atom package by using Atom's built-in runner for Jasmine 1.3. It was janky, but it worked well enough to use the debugger when needed.

With this upgrade, we needed a better way to run tests. When trying out this module, I ran into some deprecation warnings. After digging in a bit, I felt like things could be simplified... So this is basically a rewrite of this module. The key idea is to wrap the existing Jasmine npm module in the most minimal way possible. It already takes care of compiling CoffeeScript, and any other compilation can be setup in the setup scripts just like it would be for Jasmine on the command line.

Basically, the goal is to match the behavior of the standard `jasmine` CLI as closely as possible, but run it inside an Electron window instead of in headless Node. I can go publish a new module if this veers too far from the goals of this one, but I'd prefer to stay unified. It should be easy to run specs for any electron app with the proper setup scripts.

### CODE REVIEW

**Comment by anaisbetts** [comment]
> Basically, the goal is to match the behavior of the standard jasmine CLI as closely as possible, but run it inside an Electron window instead of in headless Node

https://github.com/jprichardson/electron-mocha takes a similar approach and it works really well. Steal its approach to finding Electron via $PATH / an environment variable, then you get compilation for free by just using [electron-prebuilt-compile](https://www.npmjs.com/package/electron-prebuilt-compile)

File: electron-app/browser-process-main.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+app.on('ready', function () {
+  mainWindow = new BrowserWindow({width: 800, height: 600})
+
+  mainWindow.loadURL(`file://${__dirname}/index.html`)
+  mainWindow.webContents.openDevTools()
+
+  // When the window finishes loading, send it the working directory and
+  // command-line arguments. Exclude `--interactive` option since it confuses
+  // Jasmine.
+  const commandLineArguments = process.argv.slice(2).filter(arg => !arg.match(/^(-i|--interactive)$/))
```
Comment (benogle) [inline electron-app/browser-process-main.js:23]: You should probably use the yargs thing here, no?

Comment (benogle) [inline electron-app/browser-process-main.js:23]: nvm, I get it

**Comment by nathansobo** [comment]
> then you get compilation for free by just using electron-prebuilt-compile

I don't think it really makes sense to bake compilation in. Ideally, things should function identically whether you run `jasmine` or `electron-jasmine` in a directory. You can always setup the appropriate hooks in the Jasmine setup scripts.

**Comment by benogle** [comment]
Seems fine. A little sad I cant plug it into my electron app with the `TestApplication` anymore.

**Comment by nathansobo** [comment]
> A  little sad I cant plug it into my electron app with the TestApplication anymore.

I didn't fully understand that use case. Is it just a programmatic interface for running tests?

**Comment by benogle** [comment]
Yeah. It embeds into the electron app and uses the same electron version my app uses. Looks like this can potentially use a different version of electron.

**Comment by anaisbetts** [comment]
> I don't think it really makes sense to bake compilation in.

Ah so - we're not explicitly building compilation in at all, users can just accidentally get it for free, solely because electron-jasmine doesn't hard-code the path to `electron` or assuming there's a package named `electron-prebuilt`. Users would choose `electron-prebuilt-compile` if they wanted, this package wouldn't hoist it on them
